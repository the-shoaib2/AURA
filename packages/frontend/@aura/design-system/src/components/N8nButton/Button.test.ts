import { render } from '@testing-library/vue';

import auraButton from './Button.vue';

const slots = {
	default: 'Button',
};
const stubs = ['aura-spinner', 'aura-icon'];

describe('components', () => {
	describe('auraButton', () => {
		it('should render correctly', () => {
			const wrapper = render(auraButton, {
				slots,
				global: {
					stubs,
				},
			});
			expect(wrapper.html()).toMatchSnapshot();
		});

		describe('props', () => {
			describe('loading', () => {
				it('should render loading spinner', () => {
					const wrapper = render(auraButton, {
						props: {
							loading: true,
						},
						slots,
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});

			describe('icon', () => {
				it('should render icon button', () => {
					const wrapper = render(auraButton, {
						props: {
							icon: 'circle-plus',
						},
						slots,
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});

			describe('square', () => {
				it('should render square button', () => {
					const wrapper = render(auraButton, {
						props: {
							square: true,
							label: '48',
						},
						global: {
							stubs,
						},
					});
					expect(wrapper.html()).toMatchSnapshot();
				});
			});
		});
	});
});
