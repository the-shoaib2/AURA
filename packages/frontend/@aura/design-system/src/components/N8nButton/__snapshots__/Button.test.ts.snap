// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > auraButton > props > icon > should render icon button 1`] = `"<button class="button button primary medium withIcon" aria-live="polite"><span class="icon"><aura-icon-stub icon="circle-plus" size="medium" spin="false"></aura-icon-stub></span>Button</button>"`;

exports[`components > auraButton > props > loading > should render loading spinner 1`] = `"<button class="button button primary medium loading withIcon" disabled="" aria-busy="true" aria-live="polite"><span class="icon"><aura-spinner-stub size="medium" type="dots"></aura-spinner-stub></span>Button</button>"`;

exports[`components > auraButton > props > square > should render square button 1`] = `
"<button class="button button primary medium square" aria-live="polite">
  <!--v-if--><span>48</span>
</button>"
`;

exports[`components > auraButton > should render correctly 1`] = `
"<button class="button button primary medium" aria-live="polite">
  <!--v-if-->Button
</button>"
`;
